httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java

```diff
@@ -153,13 +153,19 @@ HttpUriRequest toHttpUriRequest(Request request, Request.Options options) throws
   private ContentType getContentType(Request request) {
     ContentType contentType = ContentType.DEFAULT_TEXT;
     for (Map.Entry<String, Collection<String>> entry : request.headers().entrySet())
-    if (entry.getKey().equalsIgnoreCase("Content-Type")) {
-      Collection values = entry.getValue();
-      if (values != null && !values.isEmpty()) {
-        contentType = ContentType.create(entry.getValue().iterator().next(), request.charset());
-        break;
+      if (entry.getKey().equalsIgnoreCase("Content-Type")) {
+        Collection<String> values = entry.getValue();
+        if (values != null && !values.isEmpty()) {
+          String mimeType = values.iterator().next();
```

Comment:
ok I figured out what's going wrong here. We are blindly throwing the entire content-type header into the mimeType field. The IllegalArgumentException is present because of the ';' which is invalid. AKA we have a bug in this code. catching the exception doesn't improve anything and it causes confusion.

The correct behavior is to parse the charset. Ex. behavior like this..

```java
String unparsed = values.iterator().next();
if (unparsed.indexOf(';') != -1) { // use charset from the response
  String mimeType = get part before ';'
  String charSet = get part after ';'
  contentType = ContentType.create(mimeType, charSet);
} else { // use charset from the request instead.
  contentType = ContentType.create(unparsed, request.charset());
}
```